SSA/Projects/FullyHomomorphicEncryption/Syntax.lean

```diff
@@ -105,11 +91,39 @@ def ROfZComputable_impl (z : ℤ) : R q n :=
   }
   R.fromPoly p
 
+/-#
+Explanation of what's going on:
+- We want to have a `cstComputable : Z -> R`, which achieves three properties:
+1. It is computable, since the meta code needs it to be computable for `[fhe_icom|..]` to elaborate.
+2. It is equal to the noncomputable definition `(coe z : R q n)`
+3. It does not unfold when using `simp_peephole`, so we can rewrite using the equation
+     `ROfZComputable_def` after calling `simp_peephole`.
+
+What we really want is to have the following:
+
+```lean
 @[implemented_by ROfZComputable_impl]
-axiom ROfZComputable_stuck_term (q n : Nat) (z : ℤ) : R q n
+irreducible_noncomputable_def ROfZComputable_stuck_term (q n : Nat) (z : ℤ) :=
```

Comment:
```suggestion
noncomputable irreducible_def ROfZComputable_stuck_term (q n : Nat) (z : ℤ) :=
```
`irreducible_noncomputable_def` is not a thing, but `irreducible_def` does take modifiers just like a normal `def`